File: ckgame-java/src/main/java/com/ckgame/politics/Scheme.java
package com.ckgame.politics;

import com.ckgame.core.calendar.GameDate;

import java.util.ArrayList;
import java.util.List;

/**
 * 阴谋对象。
 */
public final class Scheme {
    public final int id;
    public final SchemeKind kind;
    public final int owner;
    public final int target;
    public double progress = 0.0;
    public double secrecy = 50.0;
    public final List<Integer> agents = new ArrayList<>();
    public GameDate started = null;
    public boolean exposed = false;

    public Scheme(int id, SchemeKind kind, int owner, int target) {
        this.id = id;
        this.kind = kind;
        this.owner = owner;
        this.target = target;
    }

    /** 阴谋是否已完成。 */
    public boolean isComplete() {
        return progress >= 100.0;
    }
}

File: ckgame-ts/src/politics/Scheme.ts
import { GameDate } from '../core/calendar/GameDate.js';
import { SchemeKind } from './SchemeKind.js';

/**
 * A scheme (conspiracy) object.
 */
export class Scheme {
  public readonly id: number;
  public readonly kind: SchemeKind;
  public readonly owner: number;
  public readonly target: number;
  public progress: number = 0;
  public secrecy: number = 50;
  public agents: number[] = [];
  public started: GameDate | null = null;
  public exposed: boolean = false;

  constructor(id: number, kind: SchemeKind, owner: number, target: number) {
    this.id = id;
    this.kind = kind;
    this.owner = owner;
    this.target = target;
  }

  /** Whether the scheme has completed. */
  isComplete(): boolean {
    return this.progress >= 100;
  }
}
